delira/utils/config.py

```diff
@@ -513,6 +515,135 @@ def create_from_str(cls, data, formatter=yaml.load, decoder_cls=Decoder,
                      **kwargs)
         return config
 
+    def create_argparser(self):
+        '''
+        Creates an argparser for all values in the config
+        Following the pattern: `--training.learning_rate 1234`
+
+        Returns
+        -------
+        argparse.ArgumentParser
+            parser for all variables in the config
+        '''
+        parser = argparse.ArgumentParser(allow_abbrev=False)
+
+        def str2bool(v):
+            if v.lower() in ('yes', 'true', 't', 'y', '1'):
+                return True
+            elif v.lower() in ('no', 'false', 'f', 'n', '0'):
+                return False
+            else:
+                raise argparse.ArgumentTypeError('Boolean value expected.')
+
+        def add_val(dict_like, prefix=''):
+            for key, val in dict_like.items():
+                name = "--{}".format(prefix + key)
+                if val is None:
+                    parser.add_argument(name)
+                else:
+                    if type(val) == bool:
+                        parser.add_argument(name, type=str2bool)
+                    elif isinstance(val, (list, tuple)):
+                        if len(val) > 0 and type(val[0]) != type:
+                            parser.add_argument(name, type=type(val[0]))
+                        else:
+                            parser.add_argument(name)
+                    elif isinstance(val, dict):
+                        add_val(val, prefix=key + '.')
+                    elif isinstance(val, type):
```

Comment:
can you change this one to `elif isinstance(val, type) of inspect.isclass(val)`? There are some few cases, where classes don't inherit from `type`

Reply:
I guess you mean `or` instead `of` right ?

Reply:
Right, that's a typo